sktime/transformations/panel/augmenter.py

```diff
@@ -0,0 +1,1217 @@
+
+    def _check_specific_aug_parameter(self):
+        """Check subclass-specific parameter (default method)."""
+        if self._param_desc is not None:
+            if self.param is None:
+                self.param = self._param_desc["default"]
+            elif isinstance(self.param, random_Variable):
+                pass
+            elif not self._param_desc["min"] <= self.param <= self._param_desc["max"]:
+                raise ValueError("Input value for param is out of boundaries.")
+        elif (
+            not isinstance(self.param, (int, float, random_Variable))
+            and self.param is not None
+        ):
+            raise TypeError(
+                f"Type of input value param must be int, float or "
+                f"a distribution of these, not {type(self.param)}."
+            )
+
+    def _plot_augmentation_example(self, X, y):
+        """Plot original and augmented instance examples for each variable.
+
+        This is a wrapper function calling the static function
+        plot_augmentation_example() for compatibility reasons.
+        """
+        plot_augmentation_example(self, X, y)
+
+
+class SeqAugPipeline(Pipeline):
+    """Subclass of `sklearn.pipeline.Pipeline`, adding functionality [1].
+
+    References
+    ----------
+    .. [1] https://scikit-learn.org/stable/modules/generated/sklearn
+       .pipeline.Pipeline.html
+
+    Examples
+    --------
+    >>> from sktime.transformations.panel.augmenter import *
+    >>> # create simple panel dataset with 2 variables and 3 instances
+    >>> X = pd.DataFrame([[pd.Series([0,1,2,3])] * 2] * 3)
+    >>> # set up Pipeline
+    >>> pipe = SeqAugPipeline(
+    ...     [
+    ...         ("invert", InvertAugmenter(p=0.5)),
+    ...         ("reverse", ReverseAugmenter(p=0.5)),
+    ...     ]
+    ... )
+    >>> Xt = pipe.fit_transform(X)
+    >>> # get information about the augmentations' random decisions
+    >>> # print(pipe.get_last_aug_random_variates())
+    """
+
+    def __init__(self, steps, memory=None, verbose=False):
+        super().__init__(steps, memory=memory, verbose=verbose)
+
+    def get_last_aug_random_variates(self):
+        """Info about last augmentation from each transformer in pipeline.
+
+        Returns
+        -------
+        list: List of pd.DataFrames of shape [n_instances, n_vars] for each
+            augmenter in the pipeline (other transformers or estimators in
+            `steps` are ignored). If an augmentation took place during the
+            last transformation call, the cell value is the used (random)
+            parameter (or None, if no parameter is needed), otherwise it is
+            False.
+        """
+        list_of_aug_info = []
+        for aug in self.steps:
+            if isinstance(aug, _BasePanelAugmenter):
+                list_of_aug_info.append(aug[1]._last_aug_random_variate)
+        return list_of_aug_info
+
+    @staticmethod
+    def draw_random_samples(
```

Comment:
Any reason this should be a part of the class rather than a standalone function?

Reply:
I fixed that.